pkg/orchestrator/pod.go

```diff
@@ -65,19 +62,49 @@ func ProcessPodList(podList []*v1.Pod, groupID int32, hostName string, clusterID
 		// additional tags
 		podModel.Tags = append(tags, fmt.Sprintf("pod_status:%s", strings.ToLower(podModel.Status)))
 
+		// The resource version field collected from the Kubelet can't be
+		// trusted because it's not updated, therefore not reflecting changes in
+		// the pod manifest.
+		// Compute our own resource version by calculating a hash of the pod
+		// model content. We'll use this information in place of the Kubelet
+		// resource version in the payload and for cache interactions.
+
+		// Enforce order consistency on slices.
+		sort.Strings(podModel.Metadata.Annotations)
+		sort.Strings(podModel.Metadata.Labels)
+		sort.Strings(podModel.Tags)
+
+		// Marshal the pod message to JSON.
+		// We need to enforce order consistency on underlying maps as
+		// the standard library does.
+		marshaller := jsoniter.ConfigCompatibleWithStandardLibrary
+		jsonPodModel, err := marshaller.Marshal(podModel)
```

Comment:
I was wondering how long/how much cpu everything would take.
Is there a noticeable increase?

I am thinking about every 10 seconds ~100 pods for one agent (max case)

For me this sounds reasonable and shouldn't be any noticeable impact. 🤔

Reply:
Given the official [Kubernetes scaling considerations](https://kubernetes.io/docs/setup/best-practices/cluster-large/) the upper bound is around 100 pods per node which shouldn't lead to a noticeable impact on performance. While I didn't assess the execution time of the extra marshaling here, the hashing is execution time is between 2~3 ms per call and given the limited number of fields in the pod payload it should be fine but we'll validate this in the next agent RC for sure.